versions.inc.sh

```diff
@@ -12,45 +12,41 @@ if [ -f ./local.env ]; then
 fi
 set +o allexport
 
-function wikibase_version {
-    echo "$MEDIAWIKI_VERSION-$WMDE_RELEASE_VERSION"
-}
-
-function elasticsearch_version {
-    echo "$ELASTICSEARCH_VERSION-$WMDE_RELEASE_VERSION"
-}
-
-function wdqs_version {
-    echo "$WDQS_VERSION-$WMDE_RELEASE_VERSION"
-}
-
-function wdqs_frontend_version {
-    echo "$WMDE_RELEASE_VERSION"
-}
-
-function wdqs_proxy_version {
-    echo "$WMDE_RELEASE_VERSION"
-}
-
-function quickstatements_version {
-    echo "$WMDE_RELEASE_VERSION"
-}
-
 function image_version {
     image=$1
 
     if [ "$image" = "wikibase" ]; then
-        wikibase_version
+        echo "$WMDE_RELEASE_VERSION"
     elif [ "$image" = "elasticsearch" ]; then
-        elasticsearch_version
+        echo "$ELASTICSEARCH_VERSION"
     elif [ "$image" = "wdqs" ]; then
-        wdqs_version
+        echo "$WDQS_VERSION"
     elif [ "$image" = "wdqs-frontend" ]; then
-        wdqs_frontend_version
+        echo "$WDQS_FRONTED_VERSION"
     elif [ "$image" = "wdqs-proxy" ]; then
-        wdqs_proxy_version
+        echo "$WDQS_PROXY_VERSION"
     elif [ "$image" = "quickstatements" ]; then
-        quickstatements_version
+        echo "$QUICKSTATEMENTS_VERSION"
     fi
 }
 
+function version_tags() {
+    local image_name=$1
+    local version=$(image_version "${image_name}")
+
+    # TODO: are these really necessary for all images?
+    local tags=(
+        "$version"
+        "${version%.*}"
+        "${version%%.*}"
+        "example-${EXAMPLE_VERSION%%.*}"
+    )
+
+    # Extra tag for wikibase
+    if [[ "$image_name" == "wikibase" ]]; then
+        tags+=("${WMDE_RELEASE_VERSION}_mw-${MEDIAWIKI_VERSION}")
```

Comment:
`_${MEDIAWIKI_VERSION}` or `+${MEDIAWIKI_VERSION}`. I think its `+` as the MediaWiki version is "extra build information" added to the version. I may be wrong...

Reply:
I can change it to a `+`, but do you also mean to discard the `mw`?

Reply:
Ah, yeah that was an oversight, please do keep the `mw` :)

Reply:
It looks like we can't use `+` in the tags
```
Error parsing reference: "wikibase/elasticsearch:1.0.0+7.10.2" is not a valid repository/tag: invalid reference format
```

Reply:
Exciting. Well, it can be safely replaced with a `_` as that has no semantics in semver. It would probably be a discussion, but also we could consider just not adding this extra build information to the tagging. 

My not-too-carefully-thought-through proposal: Keep the semver scheme intact in your script, and at the point of tagging call a function which replaces all `+`s with `_`s in the tag.

Reply:
I prefer to keep the tag consistent throughout the stack to avoid confusion. But I could change the `-` to a `_` so that we don't unintentionally imply that it's a pre-release.

Reply:
Agreed.